File: js-pr/56/javascript/code-quality-readability-maintainability/config_validator_25.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function validateConfig(config) {
    let requiredFields = ['database', 'api', 'logging'];
    for (let field of requiredFields) {
        if (!config[field]) {
            return [false, `Missing required field: ${field}`];
        }
    }
    if (!config.database.host || !config.database.port) {
        return [false, "Database configuration incomplete"];
    }
    if (!config.api.baseUrl || !config.api.timeout) {
        return [false, "API configuration incomplete"];
    }
    if (!config.logging.level || !config.logging.file) {
        return [false, "Logging configuration incomplete"];
    }
    return [true, "Configuration valid"];
}
// {/fact}

File: python-pr/52/python/code-quality-readability-maintainability/config_validator_25.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def validate_config(config):
    required_fields = ['database', 'api', 'logging']
    for field in required_fields:
        if field not in config:
            return False, f"Missing required field: {field}"
    if 'host' not in config['database'] or 'port' not in config['database']:
        return False, "Database configuration incomplete"
    if 'base_url' not in config['api'] or 'timeout' not in config['api']:
        return False, "API configuration incomplete"
    if 'level' not in config['logging'] or 'file' not in config['logging']:
        return False, "Logging configuration incomplete"
    return True, "Configuration valid" 
# {/fact}